packages/ui/src/common/components/forms/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,181 @@
+import { addMonths, addWeeks, addYears, isAfter, isBefore, isEqual, startOfMonth, startOfToday } from 'date-fns'
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import { ButtonSecondary, FilterButtons } from '@/common/components/buttons'
+import { Shadows } from '@/common/constants'
+import { DateRange, PartialDateRange } from '@/common/types/Dates'
+import { earliest, fromRange, latest, toDDMMYY } from '@/common/utils/dates'
+
+import { Calendar } from '../Calendar'
+import { InputComponent, InputText } from '../InputComponent'
+
+interface DatePickerProps {
+  value: PartialDateRange
+  withinDates?: PartialDateRange
+  onApply?: () => void
+  onClear?: () => void
+  onChange: (value: PartialDateRange) => void
+}
+
+export const DatePicker = ({ value, withinDates, onApply, onClear, onChange }: DatePickerProps) => {
+  const placeholder = '__/__/__'
+  const { start, end } = fromRange(value)
+  const dateString = `${toDDMMYY(start) ?? placeholder} - ${toDDMMYY(end) ?? placeholder}`
+
+  const container = useRef<HTMLDivElement>(null)
```

Comment:
WDYT? This way you don't need to worry if it's `null` or not.
```suggestion
  const container = useRef<HTMLDivElement>(document.createElement('div'))
```

Reply:
I think this would be fine here. But I generally would prefer being aware that a reference might be `null` (at least on the first render) than having the code potentially interact with a `div` which will get discarded. In my opinion it might be especially confusing in some cases when the `ref` is passed down other components.